sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1638,6 +1660,27 @@ class Analyzer(override val catalogManager: CatalogManager)
     }
   }
 
+  /**
+   * Optionally resolve the name parts using the outer query plan and wrap resolved attributes
+   * with [[OuterReference]]s.
+   */
+  private def resolveOuterReference(
```

Comment:
Just a question; instead of using the outer reference, we cannot simply pull up a filter condition into a join condition?
For example, in a query below, can an analyzer converts`'Join LateralJoin(Inner)` into `'Join LateralJoin(Inner), 't1.c1 = c1#171`, then resolves `'t1.c1` by `ResolveReferences`?
```
 +- 'Join LateralJoin(Inner)
    :- SubqueryAlias spark_catalog.default.t1
    :  +- View (`default`.`t1`, [c1#167,c2#168])
    :     +- Project [cast(col1#169 as int) AS c1#167, cast(col2#170 as int) AS c2#168]
    :        +- LocalRelation [col1#169, col2#170]
    +- 'SubqueryAlias __auto_generated_subquery_name
       +- 'Project ['c2]
          +- 'Filter ('t1.c1 = c1#171)
             +- SubqueryAlias spark_catalog.default.t2
                +- View (`default`.`t2`, [c1#171,c2#172])
                   +- Project [cast(col1#173 as int) AS c1#171, cast(col2#174 as int) AS c2#172]
                      +- LocalRelation [col1#173, col2#174]
```

Reply:
Pulling up correlated conditions can be tricky. In this case, it works because it's a simple equality condition, but for other cases such as having correlated non-equality conditions below an aggregate, decorrelation becomes more complex, and directly pulling up the filter conditions as join conditions may yield incorrect results (SPARK-17348 and SPARK-35080).